app/src/main/java/com/es/findsoccerplayers/ActivitySetLocation.java

```diff
@@ -0,0 +1,324 @@
+                    position_fab.setBackgroundTintList(ColorStateList.valueOf(ContextCompat.getColor(getApplicationContext(), R.color.blue)));
+                    radius.setProgress(0);
+                }
+
+            }
+        });
+
+
+
+        radius.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
+            @Override
+            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
+                if(isTracking){
+                    stopTrackingLocation();
+                }
+                if(myPosition == null){
+                    Toast.makeText(ActivitySetLocation.this, "Enable GPS to set your position", Toast.LENGTH_SHORT).show();
+                } else{
+                    drawCircle(mMap, myPosition, progress * 500);
+                    if(progress <= 10){
+                        mMap.animateCamera(CameraUpdateFactory.newLatLngZoom(myPosition, 12));
+                    } else if(progress > 10 && progress < 20){
+                        mMap.animateCamera(CameraUpdateFactory.newLatLngZoom(myPosition, 11));
+                    } else if(progress >= 20 && progress < 40){
+                        mMap.animateCamera(CameraUpdateFactory.newLatLngZoom(myPosition, 10));
+                    }else {
+                        mMap.animateCamera(CameraUpdateFactory.newLatLngZoom(myPosition, 9));
+                    }
+                    i = progress * 0.5;
+                    distance.setText(i + " km");
+                }
+            }
+
+            @Override
+            public void onStartTrackingTouch(SeekBar seekBar) {
+
+            }
+
+            @Override
+            public void onStopTrackingTouch(SeekBar seekBar) {
+
+            }
+        });
+
+        save.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if(myPosition == null || i <= 0){
+                    Toast.makeText(ActivitySetLocation.this, R.string.complete_preferences, Toast.LENGTH_SHORT).show();
+                } else {
+                    latitude = String.valueOf(myPosition.latitude);
+                    longitude = String.valueOf(myPosition.longitude);
+                    distaceValue = String.valueOf(i * 1000);
+
+                    SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(getApplicationContext());
+                    SharedPreferences.Editor editor = sharedPreferences.edit();
+                    editor.putString(LATITUDE, latitude);
+                    editor.putString(LONGITUDE, longitude);
+                    editor.putString(RADIUSVALUE, distaceValue);
+
+                    editor.commit();
+                    //TODO: delete the toast and add a finish() to go back in ActivityMain
+                    Toast.makeText(ActivitySetLocation.this, "Preferences Saved!", Toast.LENGTH_SHORT).show();
+                }
+
+            }
+        });
+
+    }
+
+
+
+
```

Comment:
tutto il codice che segue è identico a quello di mapsactivity. Magari si può scrivere una classe per wrappare?

Reply:
Potrei vedere come fare, ma non so se è possibile. 
Nelle righe 82-84 la creazione della mappa è un AsyncTask che poi chiama onMapReady che è il mio callback. In realta la parte di onCreate continua in parallelo alla creazione della mappa.